packages/tangle/message.go

```diff
@@ -119,19 +159,28 @@ type Message struct {
 	signature       ed25519.Signature
 
 	// derived properties
-	id             *MessageID
-	idMutex        sync.RWMutex
-	contentID      *ContentID
-	contentIDMutex sync.RWMutex
-	bytes          []byte
-	bytesMutex     sync.RWMutex
+	id         *MessageID
+	idMutex    sync.RWMutex
+	bytes      []byte
+	bytesMutex sync.RWMutex
 }
 
 // NewMessage creates a new message with the details provided by the issuer.
-func NewMessage(parent1ID MessageID, parent2ID MessageID, issuingTime time.Time, issuerPublicKey ed25519.PublicKey, sequenceNumber uint64, payload payload.Payload, nonce uint64, signature ed25519.Signature) (result *Message) {
+func NewMessage(strongParents []MessageID, weakParents []MessageID, issuingTime time.Time, issuerPublicKey ed25519.PublicKey, sequenceNumber uint64, payload payload.Payload, nonce uint64, signature ed25519.Signature) (result *Message) {
+	// syntactical validation
+	parentsCount := len(strongParents) + len(weakParents)
```

Comment:
It would be clearer if parents were sorted and filtered before counting imho.

Reply:
you are right